server/plugins/client/bundle-handler-factory.js

```diff
@@ -0,0 +1,54 @@
+module.exports.createHandler = createHandler
+
+var fs = require('fs')
+
+var log = require('npmlog')
+var bundleClient = require('./bundle')
+
+function createHandler (hoodieClientPath, bundleTargetPath, bundleConfig) {
+  var bundlePromise
+
+  return function (request, reply) {
+    if (!bundlePromise) {
+      bundlePromise = new Promise(function (resolve, reject) {
+        log.silly('client', 'bundling')
+        bundleClient(hoodieClientPath, bundleTargetPath, bundleConfig, function (error, bundleBuffer, hasUpdate) {
+          if (error) {
+            return reject(error)
+          }
+
+          writeClientBundle(hasUpdate, bundleConfig.inMemory, bundleTargetPath, bundleBuffer)
+
+          resolve(bundleBuffer)
+        })
+      })
+    }
+
+    bundlePromise.then(function (buffer) {
+      reply(buffer).bytes(buffer.length).type('application/javascript')
+    }).catch(reply)
+  }
+}
```

Comment:
you are right, and I think we need to bundle more than just `hoodie.js` soon. We could probably combine the two, have a factory that does nothing but defining the bundle variable and then pass it into a pre-bound function like my suggestion above? So the factory method would be only two lines, and we can test the pre-bound function quite easily if we want to